R/glmClassical.R

```diff
@@ -0,0 +1,1223 @@
+
+  if (options[["vif"]])
+    jaspResults[["diagnosticsContainer"]][["multicolliTable"]]$addColumnInfo(name = "VIF", title = gettext("VIF"), type = "number")
+
+  glmFullModel <- jaspResults[["glmModels"]][["object"]][["fullModel"]]
+  .glmMulticolliTableFill(jaspResults, dataset, options, ready, glmObj = glmFullModel)
+
+}
+
+.glmMulticolliTableFill <- function(jaspResults, dataset, options, ready, glmObj) {
+  vif_obj       <- .vif.default(glmObj)
+
+  if (is.matrix(vif_obj)) {
+    var_names     <- rownames(vif_obj)
+    n_var         <- length(var_names)
+    vif_vec       <- vif_obj[,1]
+    tolerance_vec <- 1/vif_vec
+  }
+  else {
+    var_names     <- names(vif_obj)
+    n_var         <- length(var_names)
+    vif_vec       <- vif_obj
+    tolerance_vec <- 1/vif_vec
+  }
+
+  for (i in 1:n_var) {
+    jaspResults[["diagnosticsContainer"]][["multicolliTable"]]$addRows(list(var       = var_names[[i]],
+                                                                            tolerance = tolerance_vec[[i]],
+                                                                            VIF       = vif_vec[[i]]))
+  }
+}
+
+
+
+
+.glmEmm <- function(jaspResults, dataset, options, ready, position) {
+
+  if (!ready) return()
+
+  emmContainer <- createJaspContainer(title = gettext("Estimated Marginal Means and Contrast Analysis"))
+  emmContainer$position <- position
+  jaspResults[["emmContainer"]] <- emmContainer
+
+  .glmMarginalMeansTable(jaspResults, dataset, options, ready, position = 1)
+  .glmContrastsTable(jaspResults, dataset, options, ready, position = 2)
+
+  return()
+}
+
+
+# Estimated marginal means
+.glmMarginalMeansTable <- function(jaspResults, dataset, options, ready, position) {
+
+  if (!ready | (length(options[["marginalMeansVars"]]) == 0))
+    return()
+
+  if (!is.null(jaspResults[["emmContainer"]][["EMMresults"]]))
+    return()
+
+  if (is.null(jaspResults[["glmModels"]]))
+    return()
+
+  glmFullModel <- jaspResults[["glmModels"]][["object"]][["fullModel"]]
+
+  # deal with continuous predictors
+  at <- NULL
+  for (var in unlist(options[["marginalMeansVars"]])) {
+    if (var %in% options[["covariates"]]) {
+      at[[var]] <- c(mean(dataset[, var], na.rm = TRUE) + c(-1, 0, 1) * options[["marginalMeansSd"]] * sd(dataset[, var], na.rm = TRUE))
+    }
+  }
+
+  # compute the results
+  emm <- emmeans::emmeans(
+    object  = glmFullModel,
+    specs   = unlist(options[["marginalMeansVars"]]),
+    at      = at,
+    options = list(level  = options[["marginalMeansCiWidth"]]),
+    type    = if (options[["marginalMeansResponse"]]) "response"
+  )
+
+  emmTable  <- as.data.frame(emm)
+  if (options[["marginalMeansComparison"]])
+    emmTest <- as.data.frame(emmeans::test(emm, null = options[["marginalMeansComparisonWith"]]))
+
+  EMMsummary <- createJaspTable(title = gettext("Estimated Marginal Means"))
```

Comment:
```suggestion
  emmSummary <- createJaspTable(title = gettext("Estimated Marginal Means"))
```
please follow a camelCase style consistently